src/tools/cephfs/JournalScanner.cc

```diff
@@ -23,6 +23,121 @@
 #define dout_context g_ceph_context
 #define dout_subsys ceph_subsys_mds
 
+bool JournalScanner::buf_is_available(bufferlist& buf)
+{
+  if (buf.length() == 0)
+    return false;
+ 
+  bufferlist temp_buf;
+  buf.copy(0, buf.length(), temp_buf);
+  do {
+    static int sentinel_num = 0;
+    bufferlist::iterator p = temp_buf.begin();
+    static int _off = 0;
```

Comment:
Please rework the code so these static variables are not necessary. Either move the variables to `JournalScanner` or pass the variables in as arguments.